contrib/OtlpGrpc/Exporter.php

```diff
@@ -0,0 +1,285 @@
+<?php
+
+declare(strict_types=1);
+namespace OpenTelemetry\Contrib\OtlpGrpc;
+
+
+use grpc;
+use OpenTelemetry\Sdk\Trace;
+use OpenTelemetry\Trace as API;
+use OpenTelemetry\Trace\Span;
+use Opentelemetry\Proto;
+use Opentelemetry\Proto\Collector\Trace\V1;
+use Opentelemetry\Proto\Common\V1\InstrumentationLibrary;
+use Opentelemetry\Proto\Common\V1\KeyValue;
+use Opentelemetry\Proto\Common\V1\AnyValue;
+use Opentelemetry\Proto\Trace\V1\Span as CollectorSpan;
+use Opentelemetry\Proto\Trace\V1\Span\SpanKind;
+use Opentelemetry\Proto\Trace\V1\InstrumentationLibrarySpans;
+use Opentelemetry\Proto\Trace\V1\Status\StatusCode;
+use Opentelemetry\Proto\Trace\V1\Status;
+
+class Exporter implements Trace\Exporter
+{
+    /**
+     * @var string
+     */
+    private $endpointURL;
+
+    /**
+     * @var string
+     */
+    private $protocol;
+
+    /**
+     * @var string
+     */
+    private $insecure;
+
+    /**
+     * @var string
+     */
+    private $certificateFile;
+
+    /**
+     * @var array
+     */
+    private $headers;
+
+    /**
+     * @var string
+     */
+    private $compression;
+
+    /**
+     * @var int
+     */
+    private $timeout;
+    /**
+     * @var SpanConverter
+     */
+    private $spanConverter;
+    
+    /**
+    * @var bool
+    */
+    private $running = true;
+
+    /**
+     * @var ClientInterface
+     */
+
+    private $client;
+
+    /**
+     * OTLP GRPC Exporter Constructor
+     * @param string $serviceName
+     */
+    public function __construct(
+        $serviceName,
+        ClientInterface $client=null
+    ) {
+
+        // Set default values based on presence of env variable
+        $this->endpointURL = getenv('OTEL_EXPORTER_OTLP_ENDPOINT') ?: 'host.docker.internal:4317';
+        //$this->endpointURL = getenv('OTEL_EXPORTER_OTLP_ENDPOINT') ?: 'api.honeycomb.io:443';
+        $this->protocol = getenv('OTEL_EXPORTER_OTLP_PROTOCOL') ?: 'grpc';
+        $this->insecure = getenv('OTEL_EXPORTER_OTLP_INSECURE') ?: 'false';
+        $this->certificateFile = getenv('OTEL_EXPORTER_OTLP_CERTIFICATE') ?: 'none';
+        $this->headers[] = getenv('OTEL_EXPORTER_OTLP_HEADERS') ?: 'none';
+        $this->compression = getenv('OTEL_EXPORTER_OTLP_COMPRESSION') ?: 'none';
+        $this->timeout =(int) getenv('OTEL_EXPORTER_OTLP_TIMEOUT') ?: 10;
+
+
+        $opts = [
```

Comment:
I'm wondering how we should make these options for metadata more extendable for different export locations?

Reply:
These likely should be set via `OTEL_EXPORTER_OTLP_HEADERS` I believe